src/graph.h

```diff
@@ -172,6 +173,10 @@ struct Edge {
 
   void Dump(const char* prefix="") const;
 
+  // Critical time is the estimated exection time in ms of the edges
+  // forming the longest time-weighted path to the target output.
+  // This quantity is used as a priority during build scheduling.
+  // NOTE: Defaults to -1 as a marker smaller than any valid time
```

Comment:
nit: s/exection/execution time :-)

Also, it may be more useful to document the use case than the implementation (which is not totally clear here), for example something like:

```
// Critical time is an estimated execution time, in milliseconds, for all commands that lead to the generation of
// the target output. It is used by the build scheduler to optimize total build time. Should be default-initialized to -1
// as a marker smaller than any valid duration.
```

Reply:
I think that phrasing is more confusing. "all commands that lead to the generation of the target output." sounds like the sum of the entire sub-graph, not just the longest path.

Reply:
On second thought I think you are correct about the phrasing, I stand corrected :)